system/ee/ExpressionEngine/Library/CP/FileManager/Traits/FileManagerTrait.php

```diff
@@ -293,22 +293,38 @@ function ($key) {
         $data = array();
         $missing_files = false;
 
-        $destinationsToEagerLoad = [];
+        // Group files by destination and eager load first
+        $eagerLoads = array_reduce($files->asArray(), function ($carry, $file) use ($member) {
+            if ($file->UploadDestination->adapter != 'local' && $file->UploadDestination->exists()) {
+                if(!array_key_exists($file->upload_location_id, $carry)) {
+                    $carry[$file->upload_location_id] = [];
+                }
+                $carry[$file->upload_location_id][] = $file;
+            }
+
+            return $carry;
+        }, []);
+
+        foreach($eagerLoads as $files) {
+            $paths = [];
+            $uploadDestination = $files[0]->UploadDestination;
+            $files = array_filter($files, function ($file) {
+                return !$file->isDirectory();
+            });
+
+            foreach($files as $file) {
+                $paths[] = $file->getAbsolutePath();
+                $paths[] = $file->getAbsoluteManipulationPath('thumbs');
+            }
+
+            $uploadDestination->getFilesystem()->eagerLoadPaths($paths);
```

Comment:
do we need to check if `eagerLoadPath` is available method?

Reply:
The filesystem has an implementation for eagerLoadPaths() so we don't need to check here, but inside that function it checks to see whether the current adapter has an implementation before calling it